server/graphql/v1/mutations/collectives.js

```diff
@@ -631,3 +632,42 @@ export async function claimCollective(_, args, req) {
 
   return collective;
 }
+
+export async function archiveCollective(_, args, req) {
+  if (!req.remoteUser) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in to archive a collective',
+    });
+  }
+
+  const collective = await models.Collective.findByPk(args.id);
+  if (!collective) {
+    throw new errors.NotFound({
+      message: `Collective with id ${args.id} not found`,
+    });
+  }
+
+  if (!req.remoteUser.isAdmin(collective.id)) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in as a core contributor.',
+    });
+  }
+
+  let tiers = await collective.getTiers();
+  tiers = tiers.map(async tier => {
+    return await tier.destroy();
+  });
+
+  // Cancel all active subscription
+  let orders = await collective.getIncomingOrders({
+    where: { status: status.ACTIVE, [Op.and]: { status: status.PENDING } },
+    include: [{ model: models.Subscription }, { model: models.Collective, as: 'collective' }],
+  });
+
+  orders = orders.map(async order => {
```

Comment:
This is confusing:

- don't transform the orders variable into an array of promises, let's keep it an array of orders. if necessary, create another variable.